src/puppetlabs/jdbc_util/pool.clj

```diff
@@ -172,7 +183,26 @@
            {:state :starting}))
 
        (init-error [this]
-         @init-error)))))
+         @init-error)
+
+       PoolLifetime
+       (block-until-ready [this]
+         (log/info (trs "Blocking execution until db init pool has finished"))
+         (deref pool-future))
```

Comment:
If the pool is `realized` the deref will return immediately, and blocks otherwise. Are you suggesting checking to avoid the log entry if it isn't going to block?